crypto/fipsmodule/service_indicator/service_indicator.c

```diff
@@ -407,6 +407,25 @@ void PBKDF2_verify_service_indicator(const EVP_MD *evp_md, size_t password_len,
   }
 }
 
+void SSHKDF_verify_service_indicator(const EVP_MD *evp_md) {
```

Comment:
I think that since [the ACVP tests will register a certain cipher](https://pages.nist.gov/ACVP/draft-celi-acvp-kdf-ssh.html#name-property-registration), then there is an expected output of the derived key, `out_len`. So I think the service indicator should check out_len when the request is for keys and not IVs.
Moreover, based on condition 3 at the end of Sec. 5.2 of [SP 800 135 r1](https://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-135r1.pdf) for SSHKDF:
> The hash function deployed in the KDFs meets the security strength(s)
required by the cryptographic function(s) for which the keying material is
being generated. The security strengths of approved hash functions used in
KDFs can be found in SP 800-57

we may even have to match the length checks with the evp_md checks, e.g. sha256 for length 128. I am not sure, but we can ask the lab.

Looking again, this is showing in the unit tests that the IVs have a certain length independent of the MD and the encryption and integrity keys have different lengths from each other and different for each MD. We may want to dig into this more to see how they're determined, or if they're arbitrary.

Reply:
New sanity checking makes sure the `out_len` is `> 0`, and I considered making it `>= the hash length`, but I didn't think there was enough context in `SSHKDF()` to know what the purpose of the output was.

The test vectors are all straight from NIST, so I'm assuming they should be considered valid.